src/main/scala/com/fulcrumgenomics/bam/EstimatePoolingFractions.scala

```diff
@@ -163,20 +169,18 @@ class EstimatePoolingFractions
   }
 
   /** Generates an iterator over non-filtered bi-allelic SNPs where all the required samples are genotyped. */
-  def constructVcfIterator(in: VCFFileReader, intervals: Option[IntervalList], samples: Array[String]): Iterator[VariantContext] = {
-    val vcfIterator: Iterator[VariantContext] = intervals match {
+  def constructVcfIterator(in: VcfSource, intervals: Option[IntervalList], samples: Seq[String]): Iterator[Variant] = {
+    val iterator: Iterator[Variant] = intervals match {
       case None     => in.iterator
-      case Some(is) => ByIntervalListVariantContextIterator(in, is)
+      case Some(is) => is.flatMap(i => in.query(i.getContig, i.getStart, i.getEnd))
     }
 
-    val samplesAsUtilSet = CollectionUtil.makeSet(samples:_*)
-
-    vcfIterator
-      .filterNot(_.isFiltered)
-      .map(_.subContextFromSamples(samplesAsUtilSet, true))
-      .filter(v => v.isSNP && v.isBiallelic && !v.isMonomorphicInSamples)
-      .filter(_.getNoCallCount == 0)
-      .filter(v => v.getGenotypesOrderedByName.iterator.forall(gt => gt.getGQ >= this.minGenotypeQuality))
+    iterator
+      .filter(v => v.filters.isEmpty || v.filters == Variant.PassingFilters)
+      .filter(v => v.alleles.size == 2 && v.alleles.forall(a => a.value.length == 1))  // Just biallelic SNPs
+      .filter(v => samples.map(v.gt).forall(gt => gt.isFullyCalled && (this.minGenotypeQuality <= 0 || gt.get[Int]("GQ").exists(_ >= this.minGenotypeQuality))))
+      .map   (v => v.copy(genotypes=v.genotypes.filter { case (s, _) => samples.contains(s)} ))
+      .filter(v => v.gts.flatMap(_.calls).toSet.size > 1)  // Not monomorphic
```

Comment:
```suggestion
      .filter(v => v.gts.flatMap(_.calls).distinct.size > 1)  // Not monomorphic
```

Reply:
I have a core belief that `distinct` is faster

Reply:
It is _at least_ as slow.  On an iterator they're about equivalent - the underlying method builds up a `HashSet` as it traverses and then skips any elements that are already present in the set. On e.g. `Seq` it's much worse as it basically builds up a `Set` and once it's done, then generates a new `Seq` with all the elements from the `Set`.

Reply:
https://gist.github.com/Mortimerp9/365fd4f306b70b2570e6

Tried googling but no luck.

Reply:
I have no faith in that benchmark.  If you look at `Iterator.distinct` it vectors to `Iterator.distinctBy(identity)`, which looks like this:

```scala
  def distinctBy[B](f: A => B): Iterator[A] = new AbstractIterator[A] {

    private[this] val traversedValues = mutable.HashSet.empty[B]
    private[this] var nextElementDefined: Boolean = false
    private[this] var nextElement: A = _

    def hasNext: Boolean = nextElementDefined || (self.hasNext && {
      val a = self.next()
      if (traversedValues.add(f(a))) {
        nextElement = a
        nextElementDefined = true
        true
      }
      else hasNext
    })

    def next(): A =
      if (hasNext) {
        nextElementDefined = false
        nextElement
      } else {
        Iterator.empty.next()
      }
  }
```

So you can see it's adding every element to a HashSet as it iterates.  Which is what `toSet` does after you dig through 16 layers of builder nonsense.